plugins/modules/content_create_template.py

```diff
@@ -0,0 +1,203 @@
+        self._template_service = self.api_client.vcenter.vm_template.LibraryItems
+        self.result = {'changed': False}
+
+        # Get parameters
+        self.template = self.params.get('template')
+        self.host = self.params.get('host')
+        self.cluster = self.params.get('cluster')
+        self.resource_pool = self.params.get('resource_pool')
+        self.library = self.params.get('library')
+        self.folder = self.params.get('folder')
+        self.vm_name = self.params.get('vm_name')
+
+        template = self.get_library_item_from_content_library_name(self.template, self.library)
+        if template:
+            self.result['template_create_info'] = dict(
+                msg="Template '%s' already exists." % self.template,
+                template_id=template,
+            )
+            module.exit_json(**self.result)
+
+    def create_template_from_vm(self):
+        # Create template placement specs
+        placement_spec = LibraryItems.CreatePlacementSpec()
+        placement_spec.host = self.get_host_by_name(self.host)
+        placement_spec.resource_pool = self.get_resource_pool_by_name(self.resource_pool)
+        placement_spec.cluster = self.get_cluster_by_name(self.cluster)
+        placement_spec.folder = self.get_folder_by_name(self.folder)
+        create_spec = LibraryItems.CreateSpec(
+            name=self.template,
+            placement=placement_spec,
+            library=self.get_library_by_name(self.library),
+            source_vm=self.get_vm_by_name(self.vm_name),
+        )
+        template_id = ''
+        try:
+            template_id = self._template_service.create(create_spec)
+        except Error as error:
+            self.module.fail_json(msg="%s" % self.get_error_message(error))
+        except Exception as err:
+            self.module.fail_json(msg="%s" % to_native(err))
+
+        if not template_id:
+            self.result['template_create_info'] = dict(
+                msg="Template creation failed",
+            )
+            self.module.fail_json(**self.result)
+        self.result['changed'] = True
+        self.result['template_create_info'] = dict(
+            msg="Create template '%s'." % self.template,
+            template_id=template_id,
+        )
+
+
+def main():
+    argument_spec = VmwareRestClient.vmware_client_argument_spec()
+    argument_spec.update(
+        template=dict(type='str', required=True),
+        library=dict(type='str', required=True),
+        vm_name=dict(type='str', required=True),
+        host=dict(type='str'),
+        cluster=dict(type='str'),
+        resource_pool=dict(type='str'),
+        folder=dict(type='str'),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_one_of=[('host', 'resource_pool', 'cluster')],
+    )
+
+    result = {'failed': False, 'changed': False}
+    vmware_contentlib_create = VmwareContentCreateTemplate(module)
```

Comment:
what about deleting and updating template?

Reply:
Added deleting, updating support only notes and name, which is not really needed imho.